rust/otap-dataflow/crates/channel/src/error.rs

```diff
@@ -1,6 +1,9 @@
 // SPDX-License-Identifier: Apache-2.0
 
 //! Errors for the channels.
+//!
+//! Important note: It is important not to use `!Send` data types in errors (e.g. avoid using Rc) to
```

Comment:
If it's important, should we enforce this requirement? Or if enforcing it is cascading the `Send` requirement, maybe have two different types for it similar to effect handlers (send vs non-send)?

```rust
pub enum SendError<T : Send>
```

Reply:
This is something I’ll most likely revisit, as I’m not satisfied with enforcing a `Send` bound on pdata.